content/en/blog/2023/end-user-q-and-a-04.md

```diff
@@ -0,0 +1,441 @@
+Sometimes they weren't, as in the case of Statsd automatically adding something 
+they didn't care about; those could be safely ignored.
+
+For tracing, Jacob says, "I picked a service that had both internal-only traces 
+(stayed within a single service) and traces that spanned multiple services with 
+different types of instrumentation, so from Envoy to OTel to OpenTracing." 
+
+He explains, "What you want to see is that the trace before has the same 
+structure as the trace after. So I made another script that checked that those 
+structures were relatively the same and that they all had the same attributes as 
+well... That’s the point of the tracing migration – what matters is that all the 
+attributes stayed the same."
+
+### When data goes missing
+
+"The ‘why it’s missing’ stories are the really complicated ones," says Jacob. 
+Sometimes, it's as simple as forgetting "to add something somewhere," but other 
+times, there could be an upstream library that doesn't emit what you expected 
+for OTel. 
+
+He tells a story about the time he  migrated their grpc util package (which is 
+now in Go contrib) and found an issue with propagation. 
+
+"I was trying to understand what’s going wrong here. When I looked at the code – 
+this tells you how early I was doing this migration – where there was supposed to 
+be a propagator, there was just a 'TODO' from Alex Boten," he shares. "It just 
+took down our entire services’ traces in staging." 
+
+He spent some time working on it, but they in turn were waiting on something 
+else, and so on and so forth -- Jacob says there are "endless cycles of that 
+type of thing." Once he resolved the problem, he upstreamed it so that it was 
+available to the community. 
+
+"A lot of the metrics work resulted in big performance boosts for OTel metrics," 
+he says. "Like OTel Go metrics. It also has given the Stats folks some ideas 
+about how descriptive the API should be for various features. 
+So things like Views and the use of Views is something we used heavily early in 
+the migration."
+
+### Metrics Views
+
+"A Metrics View is something that is run inside of your Meter Provider in OTel," 
+Jacob explains. There are many configuration options, such as dropping 
+attributes, which is one of the most common use cases. "For example, you’re a 
+centralized SRE and you don't want anyone to instrument code with any user ID 
+attribute, because that’s a high cardinality thing and it’s going to explode 
+your metrics cost. You can make a View that gets added to your instrumentation 
+and tell it to not record it, to deny it."
+
+There are also more advanced use cases, for example, dynamically changing the 
+temporality or aggregation of your metrics. Temporality refers to whether a 
+metric incorporates the previous measurement or not (cumulative and delta), 
+and aggregation refers to how you send off the metrics. 
+
+"It’s most useful for histograms," says Jacob. "When you record histograms, 
```

Comment:
i may amend this to say `It's most useful for [our] histograms`, because I don't think a lot of other people will have this issue.